xchange-kraken/src/main/java/org/knowm/xchange/kraken/service/KrakenTradeService.java

```diff
@@ -68,32 +68,44 @@ public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
 
   /**
    * @param params Can optionally implement {@link TradeHistoryParamOffset} and {@link
-   *     TradeHistoryParamsTimeSpan}. All other TradeHistoryParams types will be ignored.
+   *     TradeHistoryParamsTimeSpan} and {@link TradeHistoryParamsIdSpan} All other
+   *     TradeHistoryParams types will be ignored.
    */
   @Override
   public UserTrades getTradeHistory(TradeHistoryParams params)
       throws ExchangeException, IOException {
 
-    final Long startTime;
-    final Long endTime;
-    if (params instanceof TradeHistoryParamsTimeSpan) {
-      TradeHistoryParamsTimeSpan timeSpan = (TradeHistoryParamsTimeSpan) params;
-      startTime = DateUtils.toUnixTimeNullSafe(timeSpan.getStartTime());
-      endTime = DateUtils.toUnixTimeNullSafe(timeSpan.getEndTime());
-    } else {
-      startTime = null;
-      endTime = null;
-    }
+    String start = null;
+    String end = null;
+
+    Long offset = null;
 
-    final Long offset;
     if (params instanceof TradeHistoryParamOffset) {
       offset = ((TradeHistoryParamOffset) params).getOffset();
-    } else {
-      offset = null;
+    }
+
+    if (params instanceof TradeHistoryParamsIdSpan) {
+      TradeHistoryParamsIdSpan idSpan = (TradeHistoryParamsIdSpan) params;
+      if (idSpan.getStartId() != null) {
```

Comment:
these `if`s are unnecessary

just doing `start = idSpan.getStartId();` for a null `idSpan.getStartId()` will have the same result in `start` as it is initialized to `null`